localstack/services/events/provider.py

```diff
@@ -523,10 +539,12 @@ def events_handler_put_events(self):
         targets = []
         for rule in matching_rules:
             if filter_event_based_on_event_format(self, rule.name, event_bus_name, formatted_event):
+                rule_targets = self.events_backend.list_targets_by_rule(
+                    rule.name, event_bus_arn(event_bus_name)
+                ).get("Targets", [])
+
                 targets.extend(
-                    self.events_backend.list_targets_by_rule(
-                        rule.name, event_bus_arn(event_bus_name)
-                    )["Targets"]
+                    [target.update({"RuleArn": rule.arn}) or target for target in rule_targets]
```

Comment:
Might be easier to use dict unions (https://peps.python.org/pep-0584/) here to achieve this, at first I did not quite get what this is supposed to mean here :)